File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.java
class Solution {
    public char nextGreatestLetter(char[] letters, char target) {
        
        int length = letters.length;
        int low = 0;
        int high = length - 1;

        while(low <= high)
        {
            int mid = (low + high) / 2;

            if(letters[mid] > target)
                high = mid - 1;
            else
                low = mid + 1;
        }

        return low == length ? letters[0] :letters[low];
    }
}

File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.js
/**
 * @param {character[]} letters
 * @param {character} target
 * @return {character}
 */
var nextGreatestLetter = function(letters, target) {
    
    let low = 0;
    let high = letters.length - 1;

    while(low <= high)
    {
        let mid = Math.floor((low + high) / 2);

        if(letters[mid] <= target)
            low = mid + 1;
        else
            high = mid - 1;
    }

    return low === letters.length ? letters[0] : letters[low];
};